packages/language-server/package.json

```diff
@@ -1,5 +1,5 @@
 {
-  "name": "hegel-language-server",
+  "name": "@hegel/language-server",
```

Comment:
VSCode Marketplace does not support scopes. We can't publish extension with such name. [Details here](https://github.com/Microsoft/vscode-vsce/issues/186).

Reply:
i am thinking to have build folder where i have bundled code with custom `package.json` that has info for `vsce` but does not contain anything that will not be used by `vsce`

so that we have two manifest files one for `vsce` and another for `yarn`/`npm` to we avoid conflict between `yarn` and `vsce` and to have cleaner `package.json` where `yarn` does not complain about engine `vscode`

`package.json` will be for `yarn` and we can have `extension.json` for `vsce` that is copied to `build/package.json` when building code and then `vsce` is run in `build` folder

Reply:
> i am thinking to have build folder where i have bundled code with custom `package.json` that has info for `vsce` but does not contain anything that will not be used by `vsce`
> 
> so that we have two manifest files one for `vsce` and another for `yarn`/`npm` to we avoid conflict between `yarn` and `vsce` and to have cleaner `package.json` where `yarn` does not complain about engine `vscode`
> 
> `package.json` will be for `yarn` and we can have `extension.json` for `vsce` that is copied to `build/package.json` when building code and then `vsce` is run in `build` folder

It can lead to duplicative code. And for updating info about extension we will change two different files that play same purpose.

Reply:
i don't have exact solution for now 

but i am not planning to have duplicated info in `extension.json` and `package.json` i am thinking to have script for generating `build/package.json` that can take some fields from `package.json` and others from `extension.json`

for example `name` will be present on both of them but `activationEvents`, `engines.vscode`, `publisher`, `displayName`, `contributes` will be in `extension.json` 

i think most of fields will be in `extension.json` and since this package is not published on npm i will make it private and remove all non essential stuff from `package.json` like `repository`, `description`, `version` that is only used when publishing to registry

Reply:
i am not sure what fields i will take from `package.json` since it's private and does not need to include anything beyond `dependencies`, `devDependencies`, `scripts` and `name` (all of those are not needed in `build/package.json`)

Reply:
@YevhenKap, I will merge it after your approved.

Reply:
i am actually still working on it 

packaged version sill does not work ...

Reply:
i have added [extension.json](https://github.com/JSMonk/hegel/pull/260/commits/b0ba540c3531114c157ef4722adbfe9a57ad6817#diff-5816054fb17d6938656ab3186824c7b3) and updated [package.json](https://github.com/JSMonk/hegel/pull/260/commits/b0ba540c3531114c157ef4722adbfe9a57ad6817#diff-13b1edd8ff3121ec540525240f3e17eb) as you can see there are no duplicated information in those two files so maintenance will not be a problem ...